view/frontend/web/js/view/payment/method-renderer/vaultAbstract.js

```diff
@@ -235,33 +235,56 @@ define([
-                );
-                this.tdsCheck.mwCallback = (liabilityShifted, tdsToken) => this._tdsCallBack(liabilityShifted, tdsToken);
-                this.tdsCheck.link(this.mwPayframe);
-                var tdsAdditionalInfo = {tokenType : "vault"};
-                this.mwPayframe.deploy();
-
-                // When you have the payframeToken and payframeKey, call checkTDS,
-                // passing in the payframeToken, payframeKey, transactionAmount, transactionCurrency
-                // and transactionProduct
-                this.tdsCheck.checkTDS(
-                    this.details.cardID,
-                    this.details.cardKey,
-                    this.getFormattedPrice(quote.totals().grand_total),
-                    quote.totals().base_currency_code,
-                    this.getItemsSku(),
-                    tdsAdditionalInfo,
-                );
+
+                let vaultClass = this;
+                let payframeEvent = function () {
+                    vaultClass.mwPayframe.loaded = () => vaultClass._payFrameLoaded();
+                    vaultClass.tdsCheck = vaultClass._initTdsCheck(
+                        vaultClass._getPaymentConfig('uuid'),
+                        vaultClass._getPaymentConfig('apiKey'),
+                        vaultClass.mwCardDivId + vaultClass.getId(),
+                        vaultClass._getPaymentConfig('submitURL'),
+                        {
+                            width: '500px',
+                            subFrame: true
+                        }
+                    );
+                    vaultClass.tdsCheck.mwCallback = (liabilityShifted, tdsToken) => vaultClass._tdsCallBack(liabilityShifted, tdsToken);
+                    vaultClass.tdsCheck.link(vaultClass.mwPayframe);
+                    var tdsAdditionalInfo = {tokenType : "vault"};
+                    vaultClass.mwPayframe.deploy();
+
+                    // When you have the payframeToken and payframeKey, call checkTDS,
+                    // passing in the payframeToken, payframeKey, transactionAmount, transactionCurrency
+                    // and transactionProduct
+                    vaultClass.tdsCheck.checkTDS(
+                        vaultClass.details.cardID,
+                        vaultClass.details.cardKey,
+                        vaultClass.getFormattedPrice(quote.totals().grand_total),
+                        quote.totals().base_currency_code,
+                        vaultClass.getItemsSku(),
+                        tdsAdditionalInfo,
+                    );
+
+                }
+                
+                let previousHeight = 0;
+                let mwIframe = this.mwPayframe.mwIframe
+                if(mwIframe){
+                    const checkLoaded = setInterval(function () {
+                        // trick for detecting laoded iframe
+                        if (mwIframe.contentDocument && mwIframe.contentDocument.body.offsetHeight !== previousHeight) {
```

Comment:
In this commit, I added two methods for identifying whether the child-iframe fully loaded or not.
I firstly periodically check the state of the iframe's content to determine if it is fully loaded. However, Practically, only applying periodically check is not sufficient for checking Iframe loading status. Thus I added a grace time (2 seconds) to ensure the child-iframe being loaded fully. 

Alternatively, we could add a pair of post Message in both parent page and child iframe to communicate the loading status.

Reply:
The reason why we cannot simply use - **_iframe.addEventListener - Load method_**;, Becasue the load event only fired once, and it already be consumed while we create the payframe. 
So if we wanna leverage iframe.addEventListener - Load method, we need to register the function while creating the payframe. It could be another option. 
Refer - 
You cannot detect if an iframe is fully loaded after it has already loaded. The load event is only fired once when the iframe is first loaded. If you missed the load event, there is no way to detect if the iframe is fully loaded at a later point in time.

Reply:
Note for myself, 
     as we never apply this solution for other products, only for Magento Multi-Payframe Subframe Vault

Reply:
It should be optimized by Commit [ea059f6](https://github.com/merchantwarrior/magento2/pull/4/commits/ea059f6657e9ab8b088396b5280929a08110e158), as a hook is being added to the consumed load event to demonstrate that the playframe are fully loaded.